tools/integration-test/src/tests/query_packet.rs

```diff
@@ -0,0 +1,149 @@
+use ibc_relayer::chain::counterparty::{channel_on_destination, pending_packet_summary};
+use ibc_relayer::link::{Link, LinkParameters};
+
+use ibc_test_framework::prelude::*;
+use ibc_test_framework::relayer::channel::query_identified_channel_end;
+use ibc_test_framework::relayer::connection::query_identified_connection_end;
+use ibc_test_framework::util::random::random_u64_range;
+
+#[test]
+fn test_query_packet_pending() -> Result<(), Error> {
+    run_binary_channel_test(&QueryPacketPendingTest)
+}
+
+pub struct QueryPacketPendingTest;
+
+impl TestOverrides for QueryPacketPendingTest {
+    fn modify_relayer_config(&self, config: &mut Config) {
+        // Disabling clear_on_start should make the relayer not
+        // relay any packet it missed before starting.
+        config.mode.packets.clear_on_start = false;
+        config.mode.packets.clear_interval = 0;
+    }
+
+    fn should_spawn_supervisor(&self) -> bool {
+        false
+    }
+
+    // Unordered channel: will permit gaps in the sequence of relayed packets
+    fn channel_order(&self) -> Order {
+        Order::Unordered
+    }
+}
+
+impl BinaryChannelTest for QueryPacketPendingTest {
+    fn run<ChainA: ChainHandle, ChainB: ChainHandle>(
+        &self,
+        _config: &TestConfig,
+        _relayer: RelayerDriver,
+        chains: ConnectedChains<ChainA, ChainB>,
+        channel: ConnectedChannel<ChainA, ChainB>,
+    ) -> Result<(), Error> {
+        let denom_a = chains.node_a.denom();
+
+        let wallet_a = chains.node_a.wallets().user1().cloned();
+        let wallet_b = chains.node_b.wallets().user1().cloned();
+
+        let amount1 = random_u64_range(1000, 5000);
+
+        info!(
+            "Performing IBC transfer with amount {}, which should *not* be relayed",
+            amount1
+        );
+
+        chains.node_a.chain_driver().transfer_token(
+            &channel.port_a.as_ref(),
+            &channel.channel_id_a.as_ref(),
+            &wallet_a.address(),
+            &wallet_b.address(),
+            amount1,
+            &denom_a,
+        )?;
+
+        sleep(Duration::from_secs(2));
```

Comment:
I've seen in other integration tests that the process sleeps for 1 second. Is there any particular rationale for having it sleep for 2 in this case?

Reply:
I had one sporadically failed run with 1 second sleep here, and none after I bumped it to 2.
It would be good to have some way to wait for confirmation.